### PR TITLE
rename tracks directly on TrackLableButton in a QLineEdit

### PR DESCRIPTION
**Pull request by BaraMGB**
renametrack

With this one we can edit the track name directly in the track. This avoids a dialog window.

### CODE REVIEW

**Comment by tresf** [comment]
👍 👍

**Comment by BaraMGB** [comment]
Have clean up the code a little bit.

**Comment by jasp00** [comment]
I get these errors when using compact track buttons:

````
QWidget::setMinimumSize: (/QLineEdit) Negative sizes (-1,0) are not possible
QWidget::setMaximumSize: (/QLineEdit) Negative sizes (-1,16777215) are not possible
````

**Comment by BaraMGB** [comment]
@jasp00 Thank you for revisiting this PR. I have exclude the resize in case of compact track buttons now.

File: src/gui/widgets/TrackLabelButton.cpp [hunk 1]
```diff
@@ -48,21 +48,23 @@ TrackLabelButton::TrackLabelButton( TrackView * _tv, QWidget * _parent ) :
...
+	if( ConfigManager::inst()->value( "ui", "compacttrackbuttons" ).toInt() )
 	{
 		setFixedSize( 32, 29 );
 	}
 	else
 	{
 		setFixedSize( 160, 29 );
+		m_renameLineEdit = new QLineEdit(this);
+		m_renameLineEdit->move( 30, (height() / 2) - ( m_renameLineEdit->sizeHint().height() / 2 ) );
+		m_renameLineEdit->setFixedWidth( this->width() - 33 );
```
Comment (jasp00) [inline src/gui/widgets/TrackLabelButton.cpp]: Why `this->width()` instead of `width()`?

**Comment by BaraMGB** [comment]
>Why this->width() instead of width()?

Updated

**Comment by Umcaruje** [comment]
@BaraMGB Should this also get reworked to use a disabled QLineEdit like #2918?

**Comment by BaraMGB** [comment]
@Umcaruje no, the button is labeled with setText().  That's no QLabel or PaintEvent.

**Comment by tresf** [comment]
^ right.  The difference here is we have a button object already that we're working from, which would make the task much more difficult.  In the case of the Mixer, it was just a label, making the consolidation a no-brainer.

One could argue to abolish the button altogether, but then the button-press behavior would have to  be painted.

@Umcaruje I had thought if that too when I made the recommendation on the Mixer and then ralized it's a button paint and not much we could do. :+1:

**Comment by midi-pascal** [comment]
Just to let you know: if a track or a fx is renamed Lmms does not consider the project as modified.
You can quit with no question about saving...

**Pull request merged**